scheduler-interface-spec.md

```diff
@@ -792,6 +792,31 @@ message ForgotAllocation {
 }
 ```
 
+#### Event Cache
+
+The Event Cache is a SchedulerPlugin that exposes events about scheduler objects aiming to help the end user to
+see these events from the shim side. An event is sent to the shim side through the callback in a form of an `EventMessage`.
+
+```protobuf
+message EventMessage {
+   enum Type {
+      REQUEST = 0;
+      APP = 1;
+      NODE = 2;
+      QUEUE = 3;
+   }
+
+   // the type of the object associated with the event
+   Type type = 1;
+   // ID of the object associated with the event
+   string ID = 2;
```

Comment:
A second thought on this ID field, maybe it is not appropriate to call this ID. 
Because ID usually means a field that is unique per record, but here it is more like an `object`, or a `resource`. Maybe `EventObject` is a better term for this?

```
EventObject {
   Type type = 0; // the type
   string name = 1; // object name
   string group = 2; // which group this object belongs to?
   map<string, string> attributes = 3; // some abitary attributes for this object (useful for selection)
}
```

In case of a request:

```
EventObject {
   Type type = REQUEST;
   string name = <<allocationKey>>;
   string group = <<applicationID>>;
}
```

in case of an application:

```
EventObject {
   Type type = APP;
   string name = <<applicationID>>;
   string group = <<queue>>;
}
```

in case of a node:

```
EventObject {
   Type type = NODE;
   string name = <<nodeName>>;
   string group = <<partition>>;
}
```

in case of a queue:

```
EventObject {
   Type type = QUEUE;
   string name = <<queuePath>>;
   string group = <<partition>>;
}
```

Reply:
Name of the object is ambiguous, could be just "application", it does not really show that it is the ID to identify just one object. It is the ID and an ID does not have to be unique. It refers to an object type with the ID, `objectID` covers it probably better. 
It also stays in line with the other objects (node and app) that we are using in other messages. The only one outside of that is the queue which now has just a name when passed with the but in the event should have the full path. `objectID` again covers it correctly because it is unique for that type.

Can you also explain the reason behind adding `group` to the event? Is that the `parent` object the event can get aggregated into? If so group did not make that clear, I had to deduce that from the reference given for the different types.

Reply:
hi @wilfred-s 

>Can you also explain the reason behind adding group to the event? Is that the parent object the event can get aggregated into? If so group did not make that clear, I had to deduce that from the reference given for the different types.

For example, if we have an event for a request and only pass the requestID, the shim side is hard to figure out which app this request belongs to. Having a parent that points to the correct app will be more efficient for the lookup.

Reply:
I think there's no disagreement (correct me if I'm wrong). I changed to PR accordingly, so please check.

Reply:
On the second thought I don't really see why we would need `map<string, string> attributes` in the `EventRecord` message. I think that can be omitted.

Reply:
Make sense. Lets keep it simple for now. Latest change LGTM